rpcserver.go

```diff
@@ -2836,6 +2837,11 @@ func extractPaymentIntent(rpcPayReq *rpcPaymentRequest) (rpcPaymentIntent, error
 		payIntent.outgoingChannelID = &rpcPayReq.OutgoingChanId
 	}
 
+	// Take cltv limit from request if set.
+	if rpcPayReq.CltvLimit != 0 {
```

Comment:
This check wouldn't be needed if we used 0 to indicate no limit.

Reply:
It is the same discussion as above. The conversion is done as early as possible to prevent having to deal with the magic value further down the call stack.